### PR TITLE
T20 - Individual Digits.

### PR DESCRIPTION
**Pull request by Priyanka15-12**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+         Console.WriteLine (DeciDigit (dec)); 
+      }
+      static string DeciDigit (decimal d) {
+         int whole = (int)d;
+         decimal frac = (d - whole);
+         int len = frac.ToString ().Length;
+         int power = (int)Math.Pow (10, (len - 2));
+         int zeroDigit = ((int)(frac * power)).ToString ().Length;
+         var res = ((len - 2) > zeroDigit) ? string.Concat (Enumerable.Repeat ("0", ((len - 2) - zeroDigit))) : "";
+         return ($"Integral part: {whole}\nFactorial part: {res}{(int)(frac * power)}");
```
Comment (MetaNaveen) [inline Program.cs:15]: Extract the individual digits as int[] from the int (whole) and frac. Use those arrays to print the Integral and Fractional part.

**Review by MetaNaveen — CHANGES_REQUESTED** [review]

File: Program.cs [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
+         Console.Write ("Enter decimal number: ");
+         decimal.TryParse (Console.ReadLine (), out decimal dec);
+         Console.Write ("Integral Part: ");
+         DeciDigit (dec);
+         Console.Write ("\nFractional Part: ");
+         Fraction (dec);
+      }
+      static void DeciDigit (decimal d) {
+         int whole = (int)d;
+         int[] wholeArr = new int[whole.ToString ().Length];
```
Comment (MetaNaveen) [inline Program.cs:13]: Using line 13-17, you could define a new method like "static int[] GetIntArrFromInt(int n) { ... }"

File: Program.cs [hunk 3]
```diff
@@ -1,7 +1,31 @@
...
+            wholeArr[i] = whole % 10;
+            whole /= 10;
+         }
+         Array.ForEach (wholeArr, i => Console.Write ($"{i} "));
+      }
+      static void Fraction (decimal d) {
+         decimal frac = (d - (int)d);
+         if (frac > 0) {
+            string fracStr = frac.ToString ().Substring (2);
+            int[] fracArr = new int[fracStr.Length];
```
Comment (MetaNaveen) [inline Program.cs:24]: Replace the line 24-26, by calling the method like "GetIntArrFromInt(fracInt)"

**Review by MetaNaveen — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 2 above)
Comment (Priyanka15-12) [reply Program.cs]: Modified.

**Comment by MetaNaveen** [comment]
Working fine.

**Final Notes:**
n.ToString ().Length is computed/used twice. So you could have used a variable and have it referenced in other places.

**Pull request closed without merging**